dozer-sql/src/pipeline/product/join.rs

```diff
@@ -840,30 +852,43 @@ impl JoinOperator {
         prefix: u32,
         database: &Database,
         transaction: &SharedTransaction,
-    ) -> Result<Vec<Vec<u8>>, ExecutionError> {
+    ) -> Result<Vec<Vec<u8>>, ProductError> {
         let mut join_keys = vec![];
 
         let mut exclusive_transaction = transaction.write();
         let right_prefix_transaction = PrefixTransaction::new(&mut exclusive_transaction, prefix);
 
-        let cursor = right_prefix_transaction.open_cursor(*database)?;
+        let cursor = right_prefix_transaction
+            .open_cursor(*database)
+            .map_err(|err| ProductError::IndexGetError(join_key.to_vec(), err))?;
 
-        if !cursor.seek(join_key)? {
+        if !cursor
+            .seek(join_key)
+            .map_err(|err| ProductError::IndexGetError(join_key.to_vec(), err))?
+        {
             return Ok(join_keys);
         }
 
         loop {
-            let entry = cursor.read()?.ok_or(ExecutionError::InternalDatabaseError(
-                StorageError::InvalidRecord,
-            ))?;
+            let entry = cursor
+                .read()
+                .map_err(|err| ProductError::IndexGetError(join_key.to_vec(), err))?;
+
+            if entry.is_none() {
+                break;
+            }
 
-            if entry.0 != join_key {
+            let (key, value) = entry.unwrap();
```

Comment:
are you sure this unwrap wont cause issues ?

Reply:
I'm doing the check at line 877 do you think 
```suggestion
            if let Some(key, value) = entry {
```
would be more readable?